django/db/backends/sqlite3/base.py

```diff
@@ -259,6 +259,9 @@ def get_new_connection(self, conn_params):
         conn.create_aggregate('VAR_POP', 1, list_aggregate(statistics.pvariance))
         conn.create_aggregate('VAR_SAMP', 1, list_aggregate(statistics.variance))
         conn.execute('PRAGMA foreign_keys = ON')
+        # The macOS bundled SQLite defaults legacy_alter_table ON, which
+        # prevents atomic table renames (feature supports_atomic_references_rename)
+        conn.execute('PRAGMA legacy_alter_table = OFF')
```

Comment:
This was the _Right Fix™_ last time I looked into this issue. 

Typically I no longer see this locally... — I shall try to reproduce, but, provisionally, should be good.